File: benchmark/corpus/python/secret_vuln_1.py
"""VULN: CWE-798 — Hardcoded API key and database password in source."""
import requests

API_KEY = "sk-proj-xK9mN2pL8qR4tY7wZ3vJ6hU1cA0dF5gB"
DATABASE_PASSWORD = "Sup3r$ecret_DB_Pass_2024!"


def fetch_data(endpoint):
    headers = {"Authorization": f"Bearer {API_KEY}"}
    response = requests.get(endpoint, headers=headers)
    return response.json()


def get_db_dsn():
    return f"postgresql://admin:{DATABASE_PASSWORD}@localhost:5432/myapp"

File: benchmark/corpus/go/secret_vuln_1.go
// VULN: CWE-798 — AWS credentials hardcoded as package-level constants
package main

import (
	"fmt"
	"net/http"
)

const (
	AWSAccessKeyID     = "AKIAIOSFODNN7EXAMPLE"
	AWSSecretAccessKey = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
	APIKey             = "sk-proj-xK9mN2pL8qR4tY7wZ3vJ6hU1cA0dF5gB"
)

func makeRequest(endpoint string) {
	req, _ := http.NewRequest("GET", endpoint, nil)
	req.Header.Set("Authorization", "Bearer "+APIKey)
	req.Header.Set("X-AWS-Access-Key", AWSAccessKeyID)
	http.DefaultClient.Do(req)
	fmt.Println("done")
}
